platform/src/main.rs

```diff
@@ -12,9 +12,10 @@ fn main() {
 
         let out = host::call_the_closure(buffer);
 
-        println!("This print statement fixes a segfault.");
-
-        std::alloc::dealloc(buffer, layout);
+        // TODO: deallocation currently causes a segfault (probably because layout doesn't match main's size).
+        // investigate why this is and then re-enable this, rather than letting the system clean up the memory.
+        //
+        // std::alloc::dealloc(buffer, layout);
```

Comment:
I have a possible idea, when we pass `buffer` in, it is being passed to `roc roc__mainForHost_0_caller`.  Could roc now own that memory and be deallocating it using roc_dealloc? 

Therefore the call to `std::alloc::dealloc(buffer);` is a double free? 

This doesn't really explain why we haven't seen this issue before.

Reply:
Additional but confusing extra context: if this `dealloc` call is left in and a `println!("...")` is added to this function (I think anywhere will work), at least locally there is no segfault. I have no idea why this is the case, but maybe this can be a clue for someone.